apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -450,6 +452,69 @@ type ApplyStrategy struct {
 	ServerSideApplyConfig *ServerSideApplyConfig `json:"serverSideApplyConfig,omitempty"`
 }
 
+// ConflictResolutionConfig describes the action used to resolve conflicts if the resource
+// to be placed already exists in the target cluster.
+type ConflictResolutionConfig struct {
+	// ActionType describes the type of the action used to resolve the conflict between the resource to be placed
+	// and the existing resource in the target cluster if they exist.
+	// +kubebuilder:default=AlwaysApply
+	// +kubebuilder:validation:Enum=AlwaysApply;Report;ApplyIfNoDiff
+	// +optional
+	ActionType ConflictResolutionActionType `json:"actionType,omitempty"`
+
+	// DiffCalculateStrategy describes how to calculate the diff between the resource on
+	// the hub cluster and the existing resources on the member cluster.
+	// This strategy is only meaningful if the action type is Report or ApplyIfNoDiff.
+	// +optional
+	DiffCalculateStrategy DiffCalculateStrategy `json:"diffCalculateStrategy,omitempty"`
+}
+
+// ConflictResolutionActionType describes the type of the action to resolve the conflict
+// if the resource to be placed already exists in the target cluster.
+// +enum
+type ConflictResolutionActionType string
+
+const (
+	// ConflictResolutionActionTypeReport will generate a report of the diff between
+	// the yaml from the hub and the resource on the member cluster based on the `diffCalculateStrategy`.
+	ConflictResolutionActionTypeReport ConflictResolutionActionType = "Report"
```

Comment:
hi ryan, what's the diff between  "report" and "applyIfNoDiff"?

do you mean for the "report", even when there's no diff, we won't apply? like a dry-run?

Another question is when the user finds there's a diff and we stop applying, how can the user tells us to continue the apply?

Reply:
yes, it's a "dryrun".  

For your second question, they need to change the "ConflictResolutionActionType" to be a different type